vault_program/src/update_vault_balance.rs

```diff
@@ -1,28 +1,78 @@
 use jito_account_traits::AccountDeserialize;
-use jito_jsm_core::loader::load_associated_token_account;
+use jito_jsm_core::loader::{load_associated_token_account, load_token_mint, load_token_program};
 use jito_vault_core::{config::Config, vault::Vault};
 use solana_program::{
-    account_info::AccountInfo, entrypoint::ProgramResult, program_error::ProgramError,
-    program_pack::Pack, pubkey::Pubkey,
+    account_info::AccountInfo, entrypoint::ProgramResult, msg, program::invoke_signed,
+    program_error::ProgramError, program_pack::Pack, pubkey::Pubkey,
 };
-use spl_token::state::Account;
+use spl_token::{instruction::mint_to, state::Account};
 
 pub fn process_update_vault_balance(
     program_id: &Pubkey,
     accounts: &[AccountInfo],
 ) -> ProgramResult {
-    let [config, vault_info, vault_token_account] = accounts else {
+    let [config, vault_info, vault_token_account, vrt_mint, vault_fee_token_account, token_program] =
+        accounts
+    else {
         return Err(ProgramError::NotEnoughAccountKeys);
     };
 
     Config::load(program_id, config, false)?;
     Vault::load(program_id, vault_info, true)?;
-    let mut vault_data = vault_info.data.borrow_mut();
-    let vault = Vault::try_from_slice_unchecked_mut(&mut vault_data)?;
-    load_associated_token_account(vault_token_account, vault_info.key, &vault.supported_mint)?;
 
-    // TODO (LB): pay out fee account for any accrued fees to vault fee wallet
-    vault.tokens_deposited = Account::unpack(&vault_token_account.data.borrow())?.amount;
+    let new_balance = Account::unpack(&vault_token_account.data.borrow())?.amount;
+
+    {
+        let vault_data = vault_info.data.borrow();
+        let vault = Vault::try_from_slice_unchecked(&vault_data)?;
+        load_associated_token_account(vault_token_account, vault_info.key, &vault.supported_mint)?;
+
+        load_token_mint(vrt_mint)?;
+        load_associated_token_account(vault_token_account, vault_info.key, &vault.supported_mint)?;
+        load_associated_token_account(vault_fee_token_account, &vault.fee_wallet, vrt_mint.key)?;
+        load_token_program(token_program)?;
+        vault.check_vrt_mint(vrt_mint.key)?;
+
+        // Calculate rewards
+
+        let reward_fee = vault.calculate_rewards_fee(new_balance)?;
```

Comment:
i think this reward fee needs to be in terms of the VRT, not the supported mint?

Reply:
Done